### ISSUE TITLE
Pipe-escape `\|` inside table is broken (also pipe-symbol in code-block)

### ISSUE THREAD

**Issue by progmem64** [docsifyjs/docsify#592]
If i have a table with an escaped pipe-symbol like the following one:

```
| h1         | h2  |
| ---        | --- |
| abc \| def | xyz |
```

it is rendered like:

![screen shot 2018-07-29 at 19 01 08](https://user-images.githubusercontent.com/18649298/43368711-d6e48904-9361-11e8-8247-668adbd3abfb.png)

because the escape-sequence is not recognized and/or handled properly.

Replacing `\|` with `&#124;` is my workaround atm.

**Comment by ddivecs** [comment]
To add onto this, there is currently no way to put a codeblock with a pipe in a table

```
|param|Type|Description|
|-----|-----|-----|
|foo|`string|function`|takes a string or function|
```

renders as 
![image](https://user-images.githubusercontent.com/5957820/44167832-3daf2c00-a09d-11e8-99ab-cf8c4c74b635.png)

and
```
|param|Type|Description|
|-----|-----|-----|
|foo|`string&#124;function`|takes a string or function|
```

renders as
![image](https://user-images.githubusercontent.com/5957820/44167878-60d9db80-a09d-11e8-85e2-497e6d222b3d.png)

currently, id have to write

```
`string`&#124;`function`
```

which is not quite what i want to write, since im trying to document a type in typescript and the whole thing is a single type.

**Comment by progmem64** [comment]
@ddivecs I ran into the same problem. In fact i am having so many problems - i already reported 2 bugs - that i restructured my docs into the original source and the docsify source. I generate the docsify source with the help of a script i wrote to fix/workaround a bunch of bugs/errors.

**Comment by ddivecs** [comment]
i ended up just inlining an HTML table because my table got a little more complex (i needed a large code block using triple back ticks instead of just an inline codeblock) but i still believe that in the simple case where you just want a vertical bar in a table, there should be an easy way to escape a vertical bar.

**Comment by kirbdee** [comment]
Looks like `marked` resolved this with 0.3.19. Could we just rebuild the scripts for CDN? Optionally you could also build docsify locally

**Comment by QingWei-Li** [comment]
`4.8.0` is released.
